swarm/src/behaviour.rs

```diff
@@ -79,16 +79,11 @@ pub(crate) type THandlerOutEvent<THandler> =
 /// it will delegate to each `struct` member and return a concatenated array of all addresses
 /// returned by the struct members.
 ///
-/// When creating a custom [`NetworkBehaviour`], you must choose one of two methods to respond to
-/// incoming events:
-/// * One option is setting a custom `out_event` with `#[behaviour(out_event = "AnotherType")]`.
-/// In this case, events generated by the custom [`NetworkBehaviour`] struct members will be
-/// converted to your custom `out_event` for you to handle after polling the swarm.
-/// * Alternatively, users that need access to the root [`NetworkBehaviour`] implementation while
-/// processing emitted events, can specify `#[behaviour(event_process = true)]` (default is false).
-/// Events generated by the behaviour's struct members are delegated to [`NetworkBehaviourEventProcess`]
-/// trait implementations. Those must be provided by the user on the type that [`NetworkBehaviour`]
-/// is derived on.
+/// Events ([`NetworkBehaviour::OutEvent`]) returned by each `struct` member are wrapped in a new
+/// `enum` event, with an `enum` variant for each `struct` member. Users can define this event
+/// `enum` themselves and provide the name to the derive macro via `#[behaviour(out_event =
+/// "MyCustomOutEvent")]`. If the user does not specify an `out_event`, the derive macro generates
+/// the event definition itself, naming it `<STRUCT_NAME>Event`.
```

Comment:
This is not true yet, as the changes of the `swarm_derive::NetworkBehaviour` proc macro are not included in this PR.

Reply:
Good catch! Thanks. Addressed in https://github.com/libp2p/rust-libp2p/pull/2784/commits/8d5f6a5098ac6eb9e12ec9376e29832c72121621.